### ISSUE TITLE
Rails 5 deprecation warning

### ISSUE THREAD

**Issue by rietta** [tinfoil/devise-two-factor#115]
```
DEPRECATION WARNING: otp_secret is not an attribute known to Active Record. This behavior is deprecated and will be removed in the next version of Rails. If you'd like otp_secret to be managed by Active Record, add `attribute :otp_secret to your class....
```

**Comment by johnam** [comment]
I get this as well as the following error:

[13] pry(main)> u.otp_secret = User.generate_otp_secret
DEPRECATION WARNING: otp_secret is not an attribute known to Active Record. This behavior is deprecated and will be removed in the next version of Rails. If you'd like otp_secret to be managed by Active Record, add `attribute :otp_secret to your class. (called from __pry__ at (pry):13)
ArgumentError: must specify a key
from ruby-2.4.2@gems/encryptor-3.0.0/lib/encryptor.rb:56:in `crypt'

**Comment by binary1101** [comment]
Is it possible that a future release of devise-two-factor will handle this itself?

**Comment by DSKonstantin** [comment]
Looks like no one wants to fix it :( 

`ArgumentError in RailsAdmin::MainController#edit`
`key must be 32 bytes`

**Comment by connorshea** [comment]
@bsedat with 5.2 coming and presumably removing the deprecated behavior, do we have any plans to resolve this?

**Comment by bsedat** [comment]
I haven't had any chance to use Rails 5.0+ unfortunately. I would welcome any contributions to help out in this regard! We probably would want to update the demo app as well.

**Comment by rogerkk** [comment]
I think this is an issue with how the attr_encrypted gem sets up the virtual attribute, and that there's a fix for this in the master branch: https://github.com/attr-encrypted/attr_encrypted/commit/00da7c96beebff5ec4f2b3c9361c92e84afafe7f

I added the master branch to my Gemfile and do not see those deprecation warnings anymore. Other deprecation warnings pop up from attr_encrypted, though.

Anyway: I don't think the problem is devise-two-factor itself, and if we are to get this fixed the target should be the attr_encrypted gem.

**Comment by rogerkk** [comment]
A new version of the attr_encrypted gem (3.1.0) was released 19 days ago. Upgrading got rid of this warning for me.

**Comment by Mifrill** [comment]
@rogerkk  thanks, it's really helpful.


add the line to Gemfile to resolve this issue:
`gem 'attr_encrypted', '~> 3.1.0'`

@bsedat Please take a look:
PR just bump the version.

besides, it was necessary to fix the travis build due `dropped support for Ruby versions 2.2 and older`, more details here:
https://docs.travis-ci.com/user/languages/ruby/#bundler-20

Thank you for your time

**Comment by Mifrill** [comment]
@mediafinger is this issue can be closed with the #139 ?
remind: https://github.com/tinfoil/devise-two-factor/pull/139#issuecomment-517281287

**Comment by bsedat** [comment]
The requested attr_encrypted version is possible to pin to within the current gemspec requirements, so you should be able to use that without any devise-two-factor updates.
